view/src/pages/home.js

```diff
@@ -0,0 +1,194 @@
+// home.js
+
+import React, { Component } from 'react';
+import axios from 'axios';
+
+import Account from '../components/account';
+import Todo from '../components/todo';
+
+import Drawer from '@material-ui/core/Drawer';
+import AppBar from '@material-ui/core/AppBar';
+import CssBaseline from '@material-ui/core/CssBaseline';
+import Toolbar from '@material-ui/core/Toolbar';
+import List from '@material-ui/core/List';
+import Typography from '@material-ui/core/Typography';
+import Divider from '@material-ui/core/Divider';
+import ListItem from '@material-ui/core/ListItem';
+import ListItemIcon from '@material-ui/core/ListItemIcon';
+import ListItemText from '@material-ui/core/ListItemText';
+import withStyles from '@material-ui/core/styles/withStyles';
+import AccountBoxIcon from '@material-ui/icons/AccountBox';
+import NotesIcon from '@material-ui/icons/Notes';
+import Avatar from '@material-ui/core/Avatar';
+import ExitToAppIcon from '@material-ui/icons/ExitToApp';
+import CircularProgress from '@material-ui/core/CircularProgress';
+
+import { authMiddleWare } from '../util/auth'
+
+const drawerWidth = 240;
+
+const styles = (theme) => ({
+	root: {
+		display: 'flex'
+	},
+	appBar: {
+		zIndex: theme.zIndex.drawer + 1
+	},
+	drawer: {
+		width: drawerWidth,
+		flexShrink: 0
+	},
+	drawerPaper: {
+		width: drawerWidth
+	},
+	content: {
+		flexGrow: 1,
+		padding: theme.spacing(3)
+	},
+	avatar: {
+		height: 110,
+		width: 100,
+		flexShrink: 0,
+		flexGrow: 0,
+		marginTop: 20
+	},
+	uiProgess: {
+		position: 'fixed',
+		zIndex: '1000',
+		height: '31px',
+		width: '31px',
+		left: '50%',
+		top: '35%'
+	},
+	toolbar: theme.mixins.toolbar
+});
+
+class home extends Component {
+	state = {
```

Comment:
I'm still very curious about this syntax.  It does not appear to be valid JavaScript.  Is this getting processed by React somehow?  Can you link me to some documentation that shows how this is used and what it does?

Reply:
Update: Andrew sent me some links, and this is apparently "public class field syntax", which "is an experimental feature provided by Babel" — so it only works if you compile your JS using Babel.  That said, it is useful for avoiding the need for extra `bind` calls, so let's keep it.